src/libs/Navigation/AppNavigator/createCustomModalStackNavigator.js

```diff
@@ -14,6 +15,12 @@ const CustomRootStackNavigator = ({
         children,
     });
     const isDisplayingModal = Boolean(_.find(descriptors, descriptor => descriptor.options.isModal));
+
+    // Store modal visible state in Onyx
+    if (isDisplayingModal) {
+        setModalVisibility(true);
```

Comment:
Something about this feels a little funky. Like we should maybe be calling this as a side effect (in `componentDidUpdate`) and not on every render of the custom navigator to be a bit more explicit about it. I know this is made more difficult by the fact that _have to use a hook_ in this component and would be unable to use it if it was a class component.

Anyways, it should be set back to `false` if we are not displaying a modal?

Reply:
> Something about this feels a little funky.

I agree, I tested previously inside a `useEffect` but I guessed we were more against adding those hooks, I am happy to add it for this use case and see what people think?

> Anyways, it should be set back to `false` if we are not displaying a modal?

I like this idea - seems safer this way. Since (as you mentioned above) I'm missing setting the modal visibility to false when a user swipes-to-dismiss the page, it's possible we could get stuck with visibility true - which could have side effects somewhere

Reply:
Nvm - no `useEffect` needed